### PR TITLE
Add Checkstyle NewlineAtEndOfFile

### PR DESCRIPTION
**Pull request by AustinShalit**

### CODE REVIEW

**Comment by SamCarlberg** [comment]
I had planned on getting this in the original checkstyle PR but it didn't work well with cross-platform stuff (code written on OS X would fail checkstyle on Windows because newlines on Unix and carriage returns on Windows)

**Comment by JLLeitschuh** [comment]
Appveyor is failing.

**Comment by AustinShalit** [comment]
Yea...  I am just testing to see if I can get this to work.

**Comment by AustinShalit** [comment]
Fixed.  Now it checks for the unix line separator at the end of files.

File: checkstyle.xml [hunk 1]
```diff
@@ -34,7 +34,9 @@
 
     <module name="SuppressWarningsFilter"/>
 
-    <!--<module name="NewlineAtEndOfFile" />--> <!-- Screwed up on Appveyor for some reason -->
+    <module name="NewlineAtEndOfFile">
+        <property name="lineSeparator" value="lf"/>
```
Comment (AustinShalit) [inline checkstyle.xml]: This can be changed to allow windows line separators also.  Thoughts?

Comment (JLLeitschuh) [inline checkstyle.xml]: I don't know. I don't understand the whole Windows line ending thing vs linux. Someone else is going to have to explain/approve this change.
@bradamiller You understand this?

Comment (AustinShalit) [inline checkstyle.xml]: Here is the related JavaDoc: http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.html

Comment (JLLeitschuh) [inline checkstyle.xml]: I think only supporting the linux line feed makes sense.

**Comment by JLLeitschuh** [comment]
LGTM!

**Pull request merged**